 2014-10-15  Ting-Wei Lan  <lantw44@gmail.com>
 
+        [GTK] Use LD_LIBRARY_PATH to make g-ir-scanner use the newly-built
+        version of libraries when running the temporary executable
+        https://bugs.webkit.org/show_bug.cgi?id=136018
+
+        Reviewed by NOBODY (OOPS!).
+
+        This patch modifies LD_LIBRARY_PATH to make sure the dynamic linker
+        find the correct version of libraries when running the temporary
+        executable to generate the .gir file.
+
+        * PlatformGTK.cmake:
+
+2014-10-15  Ting-Wei Lan  <lantw44@gmail.com>
+
         [GTK] LDFLAGS is ignored when running g-ir-scanner
         https://bugs.webkit.org/show_bug.cgi?id=136018
 
--- Source/WebKit2/PlatformGTK.cmake	
+++ Source/WebKit2/PlatformGTK.cmake	
@@ -731,11 +731,24 @@ foreach (linker_flag ${INTROSPECTION_ADDITIONAL_LINKER_FLAGS})
     endif ()
 endforeach (linker_flag)
 
+# Add ${CMAKE_LIBRARY_OUTPUT_DIRECTORY} to LD_LIBRARY_PATH
+string(COMPARE EQUAL "$ENV{LD_LIBRARY_PATH}" "" ld_library_path_not_exist)
+if (ld_library_path_does_not_exist)
+    set(INTROSPECTION_ADDITIONAL_LIBRARY_PATH
+        "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}"
+    )
+else ()
+    set(INTROSPECTION_ADDITIONAL_LIBRARY_PATH
+        "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}:$ENV{LD_LIBRARY_PATH}"
+    )
+endif ()
+
 add_custom_command(
     OUTPUT ${CMAKE_BINARY_DIR}/WebKit2-${WEBKITGTK_API_VERSION}.gir
     DEPENDS WebKit2
     DEPENDS ${CMAKE_BINARY_DIR}/JavaScriptCore-${WEBKITGTK_API_VERSION}.gir
     COMMAND CC=${CMAKE_C_COMPILER} CFLAGS=-Wno-deprecated-declarations LDFLAGS=
+        LD_LIBRARY_PATH="${INTROSPECTION_ADDITIONAL_LIBRARY_PATH}"
         ${INTROSPECTION_SCANNER}
         --quiet
         --warn-all
@@ -775,6 +788,7 @@ add_custom_command(
     DEPENDS ${CMAKE_BINARY_DIR}/JavaScriptCore-${WEBKITGTK_API_VERSION}.gir
     DEPENDS ${CMAKE_BINARY_DIR}/WebKit2-${WEBKITGTK_API_VERSION}.gir
     COMMAND CC=${CMAKE_C_COMPILER} CFLAGS=-Wno-deprecated-declarations LDFLAGS=
+        LD_LIBRARY_PATH="${INTROSPECTION_ADDITIONAL_LIBRARY_PATH}"
         ${INTROSPECTION_SCANNER}
         --quiet
         --warn-all
